my/core/discovery_pure.py

```diff
@@ -114,10 +114,31 @@ def _extract_requirements(a: ast.Module) -> Requires:
 
 # todo should probably be more defensive..
 def all_modules() -> Iterable[HPIModule]:
+    for my_root in _iter_my_roots():
+        yield from _modules_under_root(my_root)
+
+
+def _iter_my_roots() -> Iterable[Path]:
+    import my  # doesn't import any code, because of namespace package
+
+    default_root = Path(__file__).absolute().parent.parent
+
+    try:
+        paths: List[str] = list(my.__path__._path)  # type: ignore[attr-defined]
+    except Exception as e:
+        logging.exception(e)
+        yield default_root
+    else:
+        if len(paths) == 0:
```

Comment:
makes sense just in case, just curious if you know of this ever actually happening?

Reply:
I don't think it would ever be possible no

The answer to both of the questions is I was doing the same as you did in `config_ok`:

https://github.com/karlicoss/HPI/blob/master/my/core/__main__.py#L154-L159

Reply:
Haha, if only I could remember now what I was thinking! Likely just knew less about packaging.
Read up https://docs.python.org/3/reference/import.html#module-path now
Seems that if it's a package (regardless whether it's namespace or ordinary), it always has __path__, but it may be empty. It only doesn't have __path__ if it's a 'leaf' module.
So guess `my` will always have it and can get rid of try-catch, but maybe makes sense to default (or throw?) if it's empty?

Reply:
Alright, I'll remove the `_path` from here then

> or throw? if its empty?

Yeah, I guess I can remove the `default_path` altogether. If `__path__` is empty (how you'd have an empty `__path__` when youre running a file in the module I don't know), it wouldn't be able to compute anything in any case, and your installation is probably broken